### DISCUSSION TITLE
Search Box Performance

### DISCUSSION THREAD

**Discussion by bamboowonder** [directus/directus#13172] [Archive]
I find that on the majority of our tables/collections, the "search" doesn't do anything.  As in you type,  nothing happens or it is very slow to return results.   The exception to this is on very small tables like lookups.   But anything with more than say 20 columns or hundreds of records, it doesn't work. 

Postgres (and maybe knex?) has a feature where if you just select the table, it returns a comma separate list of all value in table.   ie: SELECT tablename FROM tablename;   That may work, though certainly not ideal for large tables.  

Is there anything we can do to make the Search usable for anything other than tiny tables?  

https://github.com/directus/directus/discussions/12250
mentions a flag to specify which columns should be included in the Search box.   This would be a great addition.   For many of our tables, the quick Search box really only needs to look at 1 or two columns of a 100+ column table.  Forcing the enduser to create a filter everytime is not great.

Particularly for times when a lookup value pops up the referenced table.  That is moment where a quick search really shines.

thank you for considering this.

**Comment by benhaynes** [comment]
Yes, this is a good one to have... let's consolidate the conversation here: https://github.com/directus/directus/discussions/12264

**Comment by bamboowonder** [comment]
Thanks Ben.   I believe this request is more related to the discussion #12250.  Because all I need is the basic search to point to a column or few that I've specified, that's indexed and I know will be performant.   

That is something that 80%+ users would appreciate being able to just click a few checkboxes for the columns that should be included in basic search.   

#12264 relates a separate search adapter, which is a whole other layer of complexity many users won't need.   It certainly has value, though I wonder if it would as widely used by most users?

Cheers,

**Reply by benhaynes** [reply]
Ahh, I see! I just added a comment over there... I agree with your approach! ❤️

**Comment by bamboowonder** [comment]
Hi @benhaynes , I'm curious.   
Is it possible that Directus uses an internal cache/index that could be corrupt or inaccurate?    Because on the demo of Directus Cloud it searches tables with thousands of records quite quickly.  But on our internal instance it just does nothing.  Seems odd...

**Reply by rijkvanzanten** [reply]
Hard to tell, as it's so specific to your exact setup and server specifications. Searching through  a couple thousand rows shouldn't take too long, so it could be that there's another bottleneck on the database side 🤔

**Reply by bamboowonder** [reply]
Does Directus do any caching or indexing separate of the database?    If I were to backup the .env and then nuke the Directus instance, with a fresh reinstall, I wonder if that may work...

**Reply by rijkvanzanten** [reply]
There's configurable memory/redis caching, yeah, but that isn't used for search 👍🏻

**Reply by bamboowonder** [reply]
I found the issue.   I've created a bug report for this here:  https://github.com/directus/directus/issues/13726
